controllers/helmchart_controller_test.go

```diff
@@ -697,14 +697,17 @@ var _ = Describe("HelmChartReconciler", func() {
 			When("Setting valid valuesFile attribute", func() {
 				updated := &sourcev1.HelmChart{}
 				Expect(k8sClient.Get(context.Background(), key, updated)).To(Succeed())
-				chart.Spec.ValuesFile = "./charts/helmchart/override.yaml"
+				updated.Spec.ValuesFile = "./testdata/charts/helmchart/override.yaml"
 				Expect(k8sClient.Update(context.Background(), updated)).To(Succeed())
 				got := &sourcev1.HelmChart{}
 				Eventually(func() bool {
 					_ = k8sClient.Get(context.Background(), key, got)
-					return got.Status.Artifact != nil &&
-						storage.ArtifactExist(*got.Status.Artifact)
+					return got.Status.Artifact != nil && storage.ArtifactExist(*got.Status.Artifact)
```

Comment:
```suggestion
					return got.Status.Artifact != nil && storage.ArtifactExist(*got.Status.Artifact) &&
got.Status.Artifact.Revision != updated.Status.Artifact.Revision
```
Should this wait for a different artifact revision?